ui/src/app/job-details/common/debug-icons/debug-icons.component.ts

```diff
@@ -0,0 +1,39 @@
+import {Component, Input, OnInit} from '@angular/core';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+
+import {AuthService} from '../../../core/auth.service';
+import {ResourceUtils} from "../../../shared/utils/resource-utils";
+
+@Component({
+  selector: 'jm-debug-icons',
+  templateUrl: './debug-icons.component.html',
+  styleUrls: ['./debug-icons.component.css']
+})
+export class JobDebugIconsComponent implements OnInit {
+  @Input() displayMessage: boolean;
+  @Input() message: string;
+  @Input() stdout: string;
+  @Input() stderr: string;
+  @Input() directory: string;
+
+  constructor(private authService: AuthService,
+              private sanitizer:DomSanitizer) {
+  }
+
+  ngOnInit() {
+    // mark URLs as safe
+  }
+
+  getResourceUrl(url: string): SafeUrl {
```

Comment:
Is the `SafeUrl` return type using the type system to indicate that the URL has gone through the sanitizer? Very cool!